src/modules/commander/state_machine_helper.cpp

```diff
@@ -418,15 +419,24 @@ bool set_nav_state(vehicle_status_s *status, actuator_armed_s *armed, commander_
 	const bool rc_loss_act_configured = rc_loss_act > link_loss_actions_t::DISABLED;
 	const bool rc_lost = rc_loss_act_configured && (status->rc_signal_lost);
 
-	bool is_armed = (status->arming_state == vehicle_status_s::ARMING_STATE_ARMED);
-	bool old_failsafe = status->failsafe;
+	const bool is_armed = (status->arming_state == vehicle_status_s::ARMING_STATE_ARMED);
+	const bool old_failsafe = status->failsafe;
 	status->failsafe = false;
 
 	// Safe to do reset flags here, as if loss state persists flags will be restored in the code below
 	reset_link_loss_globals(armed, old_failsafe, rc_loss_act);
 	reset_link_loss_globals(armed, old_failsafe, data_link_loss_act);
 	reset_offboard_loss_globals(armed, old_failsafe, offb_loss_act, offb_loss_rc_act);
 
+	bool general_reaction = false; ///< tell the mode specific reaction implementation that there was a general failsafe
```

Comment:
Shouldn't this be called `datalink_failsafe_active`? `general_reaction` seems a bit confusing to me.

Reply:
Thanks for having a look. The idea behind `general_reaction` is that this won't be the last failsafe that can be generalized. For the time being we can name it after data link.

Reply:
Hm, I see.